checks/frozen_deps.go

```diff
@@ -28,10 +31,76 @@ func init() {
 
 // FrozenDeps will check the repository if it contains frozen dependecies.
 func FrozenDeps(c *checker.CheckRequest) checker.CheckResult {
+	return checker.MultiCheckAnd(
+		lockFilesPresent,
+		workflowActionsPinned,
+	)(c)
+}
+
+// TODO: need to support Docker https://github.com/ossf/scorecard/issues/403
+
+// Check file content
+func onFileContent(path string, content []byte,
+	logf func(s string, f ...interface{})) bool {
+
+	// Structure for workflow config.
+	// We only retrieve what we need for logging.
+	// Github workflows format: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions
+	type YamlWorflowConfig struct {
+		Name string `yaml:name`
+		Jobs map[string]struct {
+			Name  string `yaml:name`
+			Steps []struct {
+				Name string `yaml:name`
+				Id   string `yaml:id`
+				Uses string `yaml:uses`
+			}
+		}
+	}
+
+	var workflow YamlWorflowConfig
+	err := yaml.Unmarshal(content, &workflow)
+	if err != nil {
+		panic(fmt.Errorf("!! frozen-deps - Cannot unmarshall file %v\n%v\n%v", path, content, string(content)))
```

Comment:
You're right. I hesitated. I want us to catch this error when we run in on our infra, so we can quickly fix it. I worried that if it's recoverable, we won't notice it.

Reply:
Thanks, We also have an HTTP endpoint for Scorecard. This will cause the process to terminate.

On the other side when users run the executable they shouldn't run into panic for things that are recoverable that was also my concern.

I concur with your thought process on we will ignore errors when we log. 

Why not utilize this opportunity and create a custom error https://gobyexample.com/errors so that in the higher level we can check if this is an error and react differently https://itnext.io/golang-error-handling-best-practice-a36f47b0b94c

What are your thoughts on custom errors?

Reply:
I agree we should use errors properly across the project, and this would also allow us to remove error messages scattered across the code. I did not do it in this PR because I thought [#247](https://github.com/ossf/scorecard/pull/347#discussion_r616364482) would be the one where we tackle it. 

I will check more in details the links you provided. Let me know if you think we should do this in this PR or another one.

Reply:
I agree that we should not silently fail. Which makes me realize all of our CheckFns today do not return error. Which means they either silently fail or panic, neither of which is a good solution. Will file an issue to fix this.

Reply:
Re: custom errors - agreed. Though, let's not merge that with this PR. We can have a follow up PR for that.

Reply:
+1 to followup pr, this is fine as-is.